fbw-a380x/src/systems/systems-host/systems/tcas/components/LegacyTcasComputer.ts

```diff
@@ -0,0 +1,1336 @@
+            console.log(' ================================ ');
+          });
+          console.log('TCAS: RA GENERATED: ', this.activeRa.info.callout);
+        }
+
+        if (this.activeRa.info.callout.repeat) {
+          this.soundManager.tryPlaySound(this.activeRa.info.callout.sound, true, true);
+        } else {
+          this.soundManager.tryPlaySound(this.activeRa.info.callout.sound, true, false);
+        }
+
+        const isCorrective = this.activeRa.info.type === RaType.CORRECT;
+        this.correctiveRa.setVar(isCorrective);
+        this.raType.setVar(this.activeRa.info.type2);
+        this.rateToMaintain.setVar(this.activeRa.info.rateToMaintain);
+        this.upAdvisoryStatus.setVar(this.activeRa.info.upAdvisory);
+        this.downAdvisoryStatus.setVar(this.activeRa.info.downAdvisory);
+        SimVar.SetSimVarValue('L:A32NX_TCAS_VSPEED_RED:1', 'Number', this.activeRa.info.vs.red[Limits.MIN]);
+        SimVar.SetSimVarValue('L:A32NX_TCAS_VSPEED_RED:2', 'Number', this.activeRa.info.vs.red[Limits.MAX]);
+        if (isCorrective) {
+          SimVar.SetSimVarValue('L:A32NX_TCAS_VSPEED_GREEN:1', 'Number', this.activeRa.info.vs.green[Limits.MIN]);
+          SimVar.SetSimVarValue('L:A32NX_TCAS_VSPEED_GREEN:2', 'Number', this.activeRa.info.vs.green[Limits.MAX]);
+        }
+
+        this.activeRa.hasBeenAnnounced = true;
+      }
+    }
+  }
+
+  /**
+   * Send intruder array to ND display
+   */
+  private emitDisplay(): void {
+    this.sendAirTraffic.length = 0;
+    const sentAirTraffic = this.airTraffic
+      .filter((traffic) => traffic.alive === true && traffic.isDisplayed === true)
+      .sort(
+        (a, b) =>
+          b.intrusionLevel - a.intrusionLevel ||
+          a.raTau - b.raTau ||
+          a.taTau - b.taTau ||
+          a.slantDistance - b.slantDistance,
+      );
+    // Limit number of contacts displayed to 8
+    sentAirTraffic.forEach((traffic: TcasTraffic, index) => {
+      if (this.debug) {
+        const debugTraffic = new NDTcasDebugTraffic(traffic);
+        debugTraffic.hidden = index >= TCAS.DISPLAY_MAX;
+        this.sendAirTraffic.push(debugTraffic);
+      } else {
+        if (index >= TCAS.DISPLAY_MAX) {
+          return;
+        }
+        this.sendAirTraffic.push(new NDTcasTraffic(traffic));
+      }
+    });
+    this.raTraffic.forEach((tf) => {
+      const traffic: TcasTraffic | undefined = sentAirTraffic.find((p) => p && p.ID === tf.ID);
+      if (!traffic && this.debug) {
+        console.log(`ERROR: RA ${tf.ID} NOT SENT`);
+      }
+    });
+
+    this.syncer.sendEvent('A32NX_TCAS_TRAFFIC', this.sendAirTraffic);
+  }
+
+  /**
+   * Main update loop
+   * @param _deltaTime delta time of this frame
+   */
+  update(_deltaTime: number): void {
+    this.soundManager.update(_deltaTime);
+
+    const deltaTime = this.updateThrottler.canUpdate(_deltaTime * (this.simRate || 1));
```

Comment:
It might be easier to use ClockEvents from the SDK, like `sub.on('realTime').whenChangedBy(LegcayTcasComputer.UPDATE_PERIOD_MS).handle(() => doAllTheStuff);`

Reply:
Probably best left for a rewrite I think, I want to keep testing on this to as minimal as possible.